cmd/pmm-managed-starlark/main_test.go

```diff
@@ -0,0 +1,122 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/percona/pmm/api/agentpb"
+	"github.com/stretchr/testify/require"
+
+	"github.com/percona/pmm-managed/services/checks"
+)
+
+var validQueryActionResult = []map[string]interface{}{
+	{"Value": "5.7.30-33-log", "Variable_name": "version"},
+	{"Value": "Percona Server (GPL), Release 33, Revision 6517692", "Variable_name": "version_comment"},
+	{"Value": "x86_64", "Variable_name": "version_compile_machine"},
+	{"Value": "Linux", "Variable_name": "version_compile_os"},
+	{"Value": "-log", "Variable_name": "version_suffix"},
+}
+
+func TestRunChecks(t *testing.T) {
+	testCases := []struct {
+		err           bool
+		version       uint32
+		name          string
+		script        string
+		errorMessage  string
+		stdErrContent string // first line of the stack trace
```

Comment:
What we really should check is stdout (valid JSON), stderr (probably with regex) and exit code

Reply:
> (probably with regex) 

I am gonna use `strings.Contains` since we are looking for specific errors in each case rather than a pattern